JSON1 formatter repeats comments with multiple files

#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): N/A
- My shellcheck version (`shellcheck --version` or "online"): `0.11.0`
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit *(I don't see a JSON1 output option there)*

#### Here's a snippet or screenshot that shows the problem:

`main.sh`:
```bash
. ./util.sh
ls '~/.profile'
```

`util.sh`:
```bash
ls '~/.profile'
```

#### Here's what shellcheck currently says:

```console
$ shellcheck -axf json1 -s bash main.sh | jq '.comments | length'
4
$ shellcheck -axf json1 -s bash main.sh | jq .comments
[
  {
    "file": "./util.sh",
    "line": 1,
    "endLine": 1,
    "column": 4,
    "endColumn": 16,
    "level": "warning",
    "code": 2088,
    "message": "Tilde does not expand in quotes. Use $HOME.",
    "fix": null
  },
  {
    "file": "main.sh",
    "line": 2,
    "endLine": 2,
    "column": 4,
    "endColumn": 16,
    "level": "warning",
    "code": 2088,
    "message": "Tilde does not expand in quotes. Use $HOME.",
    "fix": null
  },
  {
    "file": "./util.sh",
    "line": 1,
    "endLine": 1,
    "column": 4,
    "endColumn": 16,
    "level": "warning",
    "code": 2088,
    "message": "Tilde does not expand in quotes. Use $HOME.",
    "fix": null
  },
  {
    "file": "main.sh",
    "line": 2,
    "endLine": 2,
    "column": 4,
    "endColumn": 16,
    "level": "warning",
    "code": 2088,
    "message": "Tilde does not expand in quotes. Use $HOME.",
    "fix": null
  }
]
```

#### Here's what I wanted or expected to see:

```console
$ shellcheck -axf json1 -s bash main.sh | jq '.comments | length'
2
$ shellcheck -axf json1 -s bash main.sh | jq .comments
[
  {
    "file": "./util.sh",
    "line": 1,
    "endLine": 1,
    "column": 4,
    "endColumn": 16,
    "level": "warning",
    "code": 2088,
    "message": "Tilde does not expand in quotes. Use $HOME.",
    "fix": null
  },
  {
    "file": "main.sh",
    "line": 2,
    "endLine": 2,
    "column": 4,
    "endColumn": 16,
    "level": "warning",
    "code": 2088,
    "message": "Tilde does not expand in quotes. Use $HOME.",
    "fix": null
  }
]
```

--- a/src/ShellCheck/Formatter/JSON1.hs
+++ b/src/ShellCheck/Formatter/JSON1.hs
@@ -120,7 +120,7 @@ collectResult ref cr sys = mapM_ f groups
         let filename = sourceFile (NE.head group)
         result <- siReadFile sys (Just True) filename
         let contents = either (const "") id result
-        let comments' = makeNonVirtual comments contents
+        let comments' = makeNonVirtual (NE.toList group) contents
         deepseq comments' $ modifyIORef ref (\x -> comments' ++ x)
 
 finish ref = do

